### PR TITLE
Allow accepting user-provided-tokens in `gr.load`

### PR DESCRIPTION
**Pull request by abidlabs**
Adds a `user_provides_token` parameter to `gr.load` that creates a `Textbox` where a user can provide their own API token. This should be hopefully encourage api developers to create Gradio demos very easily and share them with other collaborators or on Spaces without the developer needing to hardcode a token.

Here's example usage:

```py
import gradio as gr
import sambanova_gradio

gr.load("Meta-Llama-3.1-70B-Instruct-8k", src=sambanova_gradio.registry, accept_token=True).launch()
```

https://github.com/user-attachments/assets/d1a79b3b-8e71-4531-93f3-2a5b2d23939e

cc @AK391 @yvrjsharma. Uses `@gr.render` under the hood -- very cool @aliabid94!

### CODE REVIEW

File: gradio/external.py [hunk 1]
```diff
@@ -64,23 +67,47 @@ def load(
...
+            else:
+                textbox = user_provides_token
+
+            @gr.render(inputs=[textbox], triggers=[textbox.submit])
+            def create(token_value):
+                if isinstance(src, Callable):
+                    return src(name, token_value, **kwargs)
+                return load_blocks_from_huggingface(
+                    name=name, src=src, hf_token=token_value, **kwargs
+                )
```
Comment (abidlabs) [inline gradio/external.py:108]: I know there is some code duplication, but extracting this out as a separate function and calling it programmatically in `gr.render()` does not work, I tried it here: https://github.com/gradio-app/gradio/pull/9807/commits/b49c64f900e625c1885c457e9502c2cdf65b8472

**Comment by pngwn** [comment]
I intensely dislike the name of this argument. Can we think of any alternatives?

**Comment by abidlabs** [comment]
I had a hard time thinking of a shorter parameter name that was also clear that it should be a boolean. What about just `user_token`?

**Review by aliabid94 — APPROVED** [review]
LGTM!

**Comment by abidlabs** [comment]
Ok renamed `user_provides_token` -> `accept_token`, and after discussion with @aliabid94, removed the ability of `accept_token` to take in a `gr.Textbox` since that's a bit premature at this point. We may want to change the UI of how users provide tokens (e.g. have an accordion that closes after the api token is provided to take up less real estate)

**Comment by abidlabs** [comment]
Will merge after CI runs

**Pull request merged**